.github/workflows/macos_test_cases.yml

```diff
@@ -5,7 +5,7 @@ on: [push, pull_request, workflow_dispatch]
 
 jobs:
     build:
-        runs-on: ["macos-latest"]
+        runs-on: ["macos-13"]
```

Comment:
Why not macos-latest? Moreover, you can avoid having two workflows for testing ubuntu and macos and merge everything to one workflow by using the following

jobs:
  build:
    strategy:
      matrix:
        os: [ubuntu-latest, macos-latest]
    runs-on: ${{ matrix.os }}